pallets/ddc-verification/src/lib.rs

```diff
@@ -2430,6 +2553,54 @@ pub mod pallet {
 			}
 		}
 
+		pub(crate) fn get_consensus_for_sub_aggregates(
+			cluster_id: &ClusterId,
+			era_id: DdcEra,
+			activities: Vec<SubAggregates>,
+			min_nodes: u16,
+			threshold: Percent,
+		) -> Result<(Vec<SubAggregates>, Vec<SubAggregates>), Vec<OCWError>> {
+			let mut customer_buckets: BTreeMap<ActivityHash, Vec<SubAggregates>> = BTreeMap::new();
+
+			// Flatten and collect all customer activities
+			for activity in activities.iter() {
+				customer_buckets
+					.entry(activity.get_consensus_id::<T>())
+					.or_default()
+					.push(activity.clone());
+			}
+
+			let mut consensus_activities = Vec::new();
+			let mut not_consensus_activities = Vec::new();
+			let mut errors = Vec::new();
+			let min_threshold = threshold * min_nodes;
```

Comment:
might be it makes sense to rename min_nodes to smth else?

min sounds like it's already threshold...